src/Netscript/killWorkerScript.ts

```diff
@@ -52,22 +52,24 @@ function killWorkerScriptByPid(pid: number): boolean {
   return false;
 }
 
-function stopAndCleanUpWorkerScript(workerScript: WorkerScript): void {
-  if (typeof workerScript.atExit === "function") {
+function stopAndCleanUpWorkerScript(ws: WorkerScript): void {
+  killNetscriptDelay(ws);
+  if (typeof ws.atExit === "function") {
     try {
-      workerScript.atExit();
+      ws.env.stopFlag = false;
+      ws.atExit();
     } catch (e: unknown) {
+      let message = e instanceof ScriptDeath ? e.errorMessage : String(e);
+      message = message.replace(/.*\|DELIMITER\|/, "");
       dialogBoxCreate(
-        `Error trying to call atExit for script ${workerScript.name} on ${workerScript.hostname} ${
-          workerScript.scriptRef.args
-        } ${String(e)}`,
+        `Error trying to call atExit for script ${[ws.name, ...ws.args].join(" ")} on ${ws.hostname}\n ${message}`,
```

Comment:
does ws.args ever contain an arg that is also objects? id assume not due to how the api works but just a thought

Reply:
Nope, they're limited to string/number/boolean, all of which cleanly convert to string.